### PR TITLE
routing: fix reverse heat_maps (to somewhere)

### PR DESCRIPTION
**Pull request by TeXitoi**
fix http://jira.canaltp.fr/browse/ITI-329

### CODE REVIEW

**Review by antoine-de — APPROVED** [review]
😆 

you can maybe add an integration test on the reverse heat_map

**Comment by TeXitoi** [comment]
Everything should be OK now.

File: source/jormungandr/tests/heat_maps_tests.py [hunk 1]
```diff
@@ -34,6 +34,17 @@
 from jormungandr import app
 from nose.tools import eq_
 
+def get_duration(coord, response):
+    lon, lat = coord.split(';')
+    lat = float(lat)
+    lon = float(lon)
+    heat_map = response['heat_maps'][0]
+    i = next((i for i, h in enumerate(heat_map['heat_matrix']['line_headers'])
+              if h['cell_lat']['min_lat'] <= lat and lat <= h['cell_lat']['max_lat']))
```
Comment (xlqian) [inline source/jormungandr/tests/heat_maps_tests.py:43]: maybe `assert i`?

File: source/routing/heat_map.cpp [hunk 2]
```diff
@@ -202,14 +204,16 @@ HeatMap fill_heat_map(const BoundBox& box,
                                                       heat_map.header[j].min_coord + height_step / 2);
                 const auto source = worker.graph[projection[i][j].source].coord;
                 const auto target = worker.graph[projection[i][j].target].coord;
-                const auto coslat = center.lat() * type::GeographicalCoord::N_DEG_TO_RAD;
+                const auto coslat = cos(center.lat() * type::GeographicalCoord::N_DEG_TO_RAD);
```
Comment (xlqian) [inline source/routing/heat_map.cpp:207]: :scream:

File: source/routing/tests/heat_map_test.cpp [hunk 3]
```diff
@@ -211,11 +210,11 @@ BOOST_AUTO_TEST_CASE(heat_map_test) {
             result.push_back(heat_map.body[i].second[j]);
         }
     }
-    BOOST_CHECK(result[0].total_seconds() == 5420);
-    BOOST_CHECK(result[1].total_seconds() == 5462);
-    BOOST_CHECK(result[2].total_seconds() == 5504);
+    BOOST_CHECK_EQUAL(result[0].total_seconds(), 236);
+    BOOST_CHECK_EQUAL(result[1].total_seconds(), 194);
+    BOOST_CHECK_EQUAL(result[2].total_seconds(), 152);
+    std::cout << result.size() << std::endl;
```
Comment (xlqian) [inline source/routing/tests/heat_map_test.cpp]: do you still want to keep this line?

File: source/jormungandr/tests/heat_maps_tests.py (hunk 1 above)
Comment (TeXitoi) [reply source/jormungandr/tests/heat_maps_tests.py:43]: `next` will throw.

**Pull request merged**